src/test/shared/defaultAwsClientBuilder.test.ts

```diff
@@ -21,12 +23,20 @@ describe('DefaultAwsClientBuilder', function () {
             const service = await builder.createAwsService(Service)
 
             assert.strictEqual(!!service.config.customUserAgent, true)
-            assert.strictEqual(
-                service.config.customUserAgent!.replace('---Insiders', ''),
-                `AWS-Toolkit-For-VSCode/testPluginVersion Visual-Studio-Code/${version}`
+            assert.ok(
+                service.config
+                    .customUserAgent!.replace('---Insiders', '')
+                    .startsWith(`AWS-Toolkit-For-VSCode/testPluginVersion Visual-Studio-Code/${version}`)
```

Comment:
Since you exposted `getClientId()`, this could keep the strictEqual() by doing `replace(getClientId(), '')`.

The benefit of strictEqual() is that test failures are more informative.